### PR TITLE
[GEOS-7990] Layer sorting index

### PR DESCRIPTION
**Pull request by PascalLike**
Hi,
some changes for supporting the layer sort by index. For details:
https://osgeo-org.atlassian.net/browse/GEOS-7990

### CODE REVIEW

**Comment by aaime** [comment]
Hum... the timing of these pull requests could not be worse... we just entered feature freeze, no new features/improvements can be added. On the bright side, the freeze is not long, two weeks (but I'd expect the core developers to start actually paying attention to new features when the final release is out, in a month)

**Comment by PascalLike** [comment]
Ok thanks, see you then!

File: src/main/src/main/java/org/geoserver/catalog/impl/ResourceInfoImpl.java [hunk 1]
```diff
@@ -79,6 +80,8 @@
     protected StoreInfo store;
     
     protected transient Catalog catalog;
+    
+    protected Integer sortIndex = ResourceInfo.DEFAULT_INDEX_FOR_NOT_SORTED;
```
Comment (aaime) [inline src/main/src/main/java/org/geoserver/catalog/impl/ResourceInfoImpl.java]: Mind, XStream deserializing will likely ignore this and set the variable to null. If you want to make sure it has a non null value you can update the CatalogImpl.resolve(...) methods to force a particular value, they are called when an object is added to the catalog (so, also during startup).

**Comment by aaime** [comment]
Hi, the freeze has been over for a while now, will there be progress on these pull requests? :-)

**Comment by PascalLike** [comment]
Excuse me if I didn't answered you before. I'll work on it as soon as possible ;-)

**Comment by aaime** [comment]
Hi, any progress? :-)

**Comment by PascalLike** [comment]
Hi aaime, excuse me, I didn't give the right attention to the issue. I found that the changes doesn't work with REST API of Geoserver. I'm trying to solve this issue, but basically I'm a little bit stuck. Do you have any suggestion? Maybe it's simpler than I imagine.

**Comment by aaime** [comment]
Can you describe what "does not work with the REST API" mean? :-)

**Comment by aaime** [comment]
Any chance of resurrecting this work? :)

**Comment by PascalLike** [comment]
Sorry that it takes so long @aaime . I'm closing it for now, it's not nice to have this PR pending for so long. I will keep the branch and I will open a new PR when I'll have elements to decide what to do with it. I apologize

**Pull request closed without merging**